temp-legacy-code/src/main/java/com/ivy/legacy/datamodel/Tag.kt

```diff
@@ -0,0 +1,39 @@
+package com.ivy.legacy.datamodel
+
+import com.ivy.data.db.entity.TagEntity
+import java.time.LocalDateTime
+import java.util.UUID
+
+data class Tag(
+    val id: UUID = UUID.randomUUID(),
+
+    val name: String,
+    val description: String? = null,
+
+    val color: Int,
+    val icon: String? = null,
+    val orderNum: Double = 0.0,
+
+    val dateTime: LocalDateTime? = null,
+    val isSynced: Boolean = false,
+    val isDeleted: Boolean = false,
+
+    val featureBit: Int = 0,
+    val featureBitValue: String? = null
+) {
+    fun toEntity(): TagEntity {
```

Comment:
Can we need put this in a mapper class or somewhere out of the data class? The motivation is because we try to keep `data classes` as data model only and no attached behavior.

Reply:
Sure,
We can do it..
I was modeling it based on current code base..
Didn't want diff practices to be present.

Reply:
Yeah, you did right 👍 Keeping consistency is a good practice! The problem is that our legacy code isn't great (I'm the one who created this tech debt in the first place 😄) and I'm actively trying to fix my mistake